scoretracker/tracker.py

```diff
@@ -1,95 +1,164 @@
-            try:
-                data = db.read()
-                database = json.loads(data)
-                if len(database) == 0:
-                    raise Exception()
-            except Exception:
-                rT['text'] = 'There are no scores yet!'
-                requests.post('https://api.groupme.com/v3/bots/post',
-                              json.dumps(rT))
-                return
-            if person is not None:
-                rT['text'] += person + ':\n'
-                if score is not None:
-                    rT['text'] += '    '+score+': '+database[person][score]+'\n'
-                else:
-                    for sco in database[person].keys():
-                        rT['text'] += '    '+sco+': '+database[person][sco]+'\n'
-            else:
-                for per, scores in database.items():
-                    rT['text'] += person + ':\n'
-                    for sco in scores:
-                        rT['text'] += '    '+sco+': '+database[per][sco]+'\n'
-    except Exception:
-        invalid_request()
-        return
-    requests.post('https://api.groupme.com/v3/bots/post', json.dumps(rT))
+    """
+    Sends to the chat the information about:
+    * All (user, score) combinations
+    * All scores of a given user
+    * A given user's score in a specific topic
+
+    @ed> /score show ron
+    @scorebot> ron:
+                   disappointment: 420
+                   understanding-the-rules: -69
+    """
+    response = {'bot_id': BOT_ID}
+    response['text'] = ''
+
+    database = get_db()
+
+    if not database:
+        # Backup did not exist or was empty
+        response['text'] = 'There are no scores yet!'
+
+    elif person is None:
+        response['text'] = '\n'.join(
+            '{person}:\n{scores}'.format(person=person, scores=get_scores(database, person))
+            for person in database
+        )
+
+    elif person not in database:
+        response['text'] = 'User is either unrecognized or has not points'
+
+    elif score is None:
+        response['text'] = get_scores(database, person, multi=False)
+
+    else:
+        response['text'] = '{person}:\n    {score}: {points}'.format(
+            person=person, score=score, points=database[person][score]
+        )
+
+    requests.post(POST_URL, json.dumps(response))
 
 
 def invalid_request():
-    rT = {}
-    rT['bot_id'] = BOT_ID
-    rT['text'] = 'That was not a valid scorebot command'
-    requests.post('https://api.groupme.com/v3/bots/post', json.dumps(rT))
+    """
+    Respond to the chat informing the user that they entered an invalid
+    command for the bot.
+    """
+    response = {
+        'bot_id': BOT_ID,
+        'text': 'That was not a valid scorebot command'
+    }
+    requests.post(POST_URL, json.dumps(response))
 
 
-# /score <person> <score> <+/-> <#>
-# /score show
-# /score show <person> [score]
 def recv_msg():
+    """
+    Handles the CRUD operations represented by the following commands:
```

Comment:
CRUD?

Reply:
Technically this does handle **create**, **read**, and **update**. Depending on how you define it, resetting someone's scores could be the same as **deletion**

Reply:
Ah, ok